### PR TITLE
Feat: Add custom redact modifier for SwiftUI

### LINKED ISSUE #4333: Session Reply SwiftUI Views Redaction
### Problem Statement

Hi,

I'm currently trying to use session reply in my app, but I've noticed that if I want to exclude certain views (SwiftUI Views) for privacy reasons, the `redactViewTypes` option is only available for classes (as it's an array of the `AnyClass` type).

Is there a workaround for this?

### Solution Brainstorm

_No response_

### Are you willing to submit a PR?

_No response_

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Added a SwiftUI modifier that allows the dev to choose with part of the View they want redacted during replay.

## :bulb: Motivation and Context

closes #4333 

## :green_heart: How did you test it?

Sample

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: Samples/iOS-Swift/iOS13-Swift/SwiftUIView.swift [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import SentrySwiftUI
 import SwiftUI
 
-struct SwiftUIView: View {
+struct SwiftUIView: View {
```
Comment (brustolin) [inline Samples/iOS-Swift/iOS13-Swift/SwiftUIView.swift:4]:
```suggestion
struct SwiftUIView: View {
```

File: Samples/iOS-Swift/iOS13-Swift/SwiftUIViewController.swift [hunk 2]
```diff
@@ -9,7 +9,7 @@ class SwiftUIViewController: UIViewController {
         
     override func viewDidLoad() {
         super.viewDidLoad()
-     
+
```
Comment (brustolin) [inline Samples/iOS-Swift/iOS13-Swift/SwiftUIViewController.swift:12]:
```suggestion
     
```

File: Sources/Swift/Tools/UIRedactBuilder.swift [hunk 3]
```diff
@@ -181,7 +180,7 @@ class UIRedactBuilder {
     
     private func mapRedactRegion(fromView view: UIView, redacting: inout [RedactRegion], rootFrame: CGRect, transform: CGAffineTransform, forceRedact: Bool = false) {
         guard !redactClassesIdentifiers.isEmpty && !view.isHidden && view.alpha != 0 else { return }
-        
+
```
Comment (brustolin) [inline Sources/Swift/Tools/UIRedactBuilder.swift:183]:
```suggestion
        
```

File: Samples/iOS-SwiftUI/iOS-SwiftUI/ContentView.swift [hunk 4]
```diff
@@ -200,6 +199,7 @@ struct ContentView: View {
                                 Text("Form Screen")
                             }
                         }
+                        .replayRedact()
```
Comment (philipphofmann) [inline Samples/iOS-SwiftUI/iOS-SwiftUI/ContentView.swift:202]: `m`: We already have `.sentryTrace`. Maybe it makes sense also to prefix this with sentry so developers see all sentry related modifies when they type `.sentry` in Xcode.

File: Sources/SentrySwiftUI/SentryReplayView.swift [hunk 5]
```diff
@@ -0,0 +1,43 @@
+#if SENTRY_NO_UIKIT
+import SentryWithoutUIKit
+#else
+import Sentry
+#endif
+import SwiftUI
+import UIKit
+
+@available(iOS 13, macOS 10.15, tvOS 13, *)
+struct SentryReplayView: UIViewRepresentable {
```
Comment (philipphofmann) [inline Sources/SentrySwiftUI/SentryReplayView.swift:10]: `m`: Can we somehow add tests for that?

File: Sources/SentrySwiftUI/SentryReplayView.swift [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    
+    /// Marks the view as containing sensitive information that should be redacted during replays.
+    ///
+    /// When this modifier is applied, any sensitive content within the view will be hidden or masked
+    /// during session replays to ensure user privacy. This is useful for views containing personal
+    /// data or confidential information that shouldn't be visible when the replay is reviewed.
+    ///
+    /// - Returns: A modifier that redacts sensitive information during session replays.
+    ///
+    func replayRedact() -> some View {
```
Comment (philipphofmann) [inline Sources/SentrySwiftUI/SentryReplayView.swift:40]: `m`: Should we mention in the comment that this is still experimental?

**Review by philipphofmann — COMMENTED** [review]
A few comments, nothing blocking?

File: Sources/SentrySwiftUI/SentryReplayView.swift (hunk 5 above)
Comment (brustolin) [reply Sources/SentrySwiftUI/SentryReplayView.swift:10]: It looks like it's possible. I will give it a try

File: Tests/SentryTests/SwiftUI/SentryRedactModifierTests.swift [hunk 7]
```diff
@@ -0,0 +1,18 @@
+#if canImport(UIKit) && canImport(SwiftUI)
+@testable import Sentry
+@testable import SentrySwiftUI
+import SwiftUI
+import XCTest
+
+class SentryRedactModifierTests: XCTestCase {
+
+    func testViewRedate() throws {
```
Comment (philipphofmann) [inline Tests/SentryTests/SwiftUI/SentryRedactModifierTests.swift:9]: `l`: This looks like a typo, or am I not getting it?

**Review by philipphofmann — APPROVED** [review]
LGTM

File: Tests/SentryTests/SwiftUI/SentryRedactModifierTests.swift (hunk 7 above)
Comment (brustolin) [reply Tests/SentryTests/SwiftUI/SentryRedactModifierTests.swift:9]: Typo. Thanks!

Comment (brustolin) [reply Tests/SentryTests/SwiftUI/SentryRedactModifierTests.swift:9]:
```suggestion
    func testViewRedacted() throws {
```

**Pull request merged**